src/scipp/core/data_group.py

```diff
@@ -0,0 +1,192 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+from __future__ import annotations
+
+import functools
+import operator
+from collections.abc import MutableMapping
+from typing import Callable, Iterable
+
+from .cpp_classes import DataArray, Dataset, Variable
+
+
+def _item_dims(item):
+    return getattr(item, 'dims', ())
+
+
+def _summarize(item):
+    if isinstance(item, DataGroup):
+        return f'{type(item).__name__}({len(item)}, {item.sizes})\n'
+    if hasattr(item, 'sizes'):
+        return f'{type(item).__name__}({item.sizes})\n'
+    return str(item)
+
+
+class DataGroup(MutableMapping):
+    """
+    A group of data. Has dims and shape, but no coords.
+    """
+
+    def __init__(self, items=None):
+        self._items = {}
+        if items is not None:
+            for name, item in items.items():
+                self[name] = item
+
+    def __len__(self) -> int:
+        return len(self._items)
+
+    def __iter__(self):
```

Comment:
You could inherit from `MutableMapping` to get a bunch of these methods for free.

Reply:
It is already inheriting from `MutableMapping` ?